src/test/unit/web_interface/rest/test_rest_firmware.py

```diff
@@ -68,13 +69,13 @@ def test_successful_uid_request(test_app):
 
 
 def test_bad_put_request(test_app):
-    result = decode_response(test_app.put('/rest/firmware'))
-    assert 'Input payload validation failed' in result['message']
+    response = test_app.put('/rest/firmware')
+    assert response.status_code == 400
 
 
-def test_submit_empty_data(test_app):
-    result = decode_response(test_app.put('/rest/firmware', data=json.dumps({})))
-    assert 'Input payload validation failed' in result['message']
+def test_submit_non_json_data(test_app):
+    response = test_app.put('/rest/firmware', data='non-json-string')
```

Comment:
I think the intention behind this test was to check that the input validation assures that a request that does not contain the required fields fails. This seems to test something different (since the input data isn't even valid JSON). What was the idea behind this change? Maybe we should test both.

Reply:
Oh okay, I understand! The issue here was that the `data` kwarg sets `Content-Type: application/x-www-form-urlencoded`. The latest versions of flask and flask-restx filter this because `application/json` is expected. In previous versions, this was no issue because the content was passed to our endpoint handler either way.

So in order to test valid json but unfulfilled scheme requirements, we should do something like:

```python
def test_submit_empty_data(test_app):
    result = decode_response(test_app.put('/rest/firmware', json={}))
    assert 'Input payload validation failed' in result['message']
```

Generally, this should also be backward-compatible to older package versions. Yet I didn't test an empty dict here, maybe we need some bogus key-value set in order to actually send content. But that would be fine, right?

Reply:
That looks good to me :+1:

Reply:
done. starting CI